nat-lab/tests/utils/command_grepper.py

```diff
@@ -0,0 +1,104 @@
+import asyncio
+from typing import List, Optional
+from utils.connection import Connection
+from utils.process import ProcessExecError
+
+
+class CommandGrepper:
+    _connection: Connection
+    _check_cmd: List[str]
+    _timeout: Optional[float]
+
+    def __init__(
+        self,
+        connection: Connection,
+        check_cmd: List[str],
+        timeout: Optional[float] = None,
+    ):
+        self._connection = connection
+        self._check_cmd = check_cmd
+        self._timeout = timeout
+
+    # Check if the expected strings are present in the output of the command.
+    # If only the primary string is provided, check for its presence in whole stdout.
+    # If secondary strings are also provided, checks, if primary and all secondaries are in one line
+    async def check_exists(
+        self, exp_primary: str, exp_secondary: Optional[List[str]] = None
+    ) -> bool:
+        try:
+            return await asyncio.wait_for(
+                self._run_check(exp_primary, exp_secondary, True),
+                self._timeout,
+            )
+        except ProcessExecError:
+            return False
+
+    # Check if the expected strings are not present in the output of the command.
+    # If only the primary string is provided, check for its absence in whole stdout.
+    # If secondary strings are also provided, checks, if primary and secondary are NOT in one line
+    async def check_not_exists(
+        self, exp: str, exp_secondary: Optional[List[str]] = None
```

Comment:
Nit: `exp` should be renamed to `exp_primary` for consistency reasons and to make the "primary" reference in the doc comment clearer

Reply:
Fixed